spicy/toolchain/src/compiler/visitors/resolver.cc

```diff
@@ -268,6 +269,17 @@ struct Visitor : public hilti::visitor::PreOrder<void, Visitor> {
     }
 
     void operator()(const type::unit::item::UnresolvedField& u, position_t p) {
+        if ( u.type() && u.type()->isA<type::Void>() && u.attributes() ) {
+            // Transparently map void field into skipping bytes fields.
+            hilti::logger().deprecated(
+                "using `void` fields with attributes is deprecated and support will be removed in a future "
+                "release; replace 'void' with 'skip bytes ...'",
+                u.meta().location());
+
```

Comment:
If I understand correctly this is inconsistent with what you did over in `spicy/toolchain/src/compiler/visitors/validator.cc` where you disallow `void` fields with attributes completely. While I would prefer a deprecation where we still allow `void` fields with attributes we also have #1427 so maybe we should just completely pull support for that so the issue wouldn't apply anymore and adjust the code here.

Reply:
This PR continues to support attributes with void for now but it does indeed deprecate them. It's handled by mapping them into `skip` fields before the validator sees them. The existing tests for that feature also continue to pass.